### PR TITLE
crud-server, fix #86: checks the presence of curi

### LINKED ISSUE #86: BUG: Resource Server API allows you update curi
- When a user adds a conduit, the curi is auto-generated 
- This curi has to be internal and unique.
- Currently the update conduit function allows user to change the curi
- Update conduit API should be updated to ignore the curi input and should not update a conduit's curi

### PR DESCRIPTION
**Pull request by kavithaii**

### CODE REVIEW

File: crud-server/src/routes/api/conduits.js [hunk 1]
```diff
@@ -76,7 +76,7 @@ router.patch('/:id', auth.required, async (req, res, next) => {
     if (!conduit) {
       return res.sendStatus(404);
     }
-    if (req.body.conduit.curi === undefined) {
+    if (!req.body.conduit.curi) {
       await conduit.update(await req.body.conduit);
       res.status(200).json({ conduit: conduit.toJSON() });
     } else {
```
Comment (m5nv) [inline crud-server/src/routes/api/conduits.js:82]: This code as written would result in the wrong status code returned....

if (!conduit || req.body.conduit.curi) {
  return res.sendStatus(404)
}

Comment (m5nv) [reply crud-server/src/routes/api/conduits.js:82]: // I stand corrected... here is a better version
if (!conduit) {
return res.sendStatus(404);
}

if (req.body.conduit.curi) {
  return res.sendStatus(400);
}

Comment (kavithaii) [reply crud-server/src/routes/api/conduits.js:82]: I have changed the code block to check for *curi* presence in API, then do the update.
if (!conduit) {
      return res.sendStatus(404);
    }
    if (req.body.conduit.curi) {
      return res.sendStatus(400);
    }

    await conduit.update(await req.body.conduit);
    res.status(200).json({ conduit: conduit.toJSON() });

**Comment by m5nv** [comment]
LGTM. Please remember to delete the remote branch after merge to master is done. Thanks!

**Pull request merged**